cernopendata/modules/theme/assets/semantic-ui/js/records/components/FileTable.js

```diff
@@ -153,11 +162,26 @@ export default function FileTable({ items, table_type, recordAvailability }) {
               </Table.Cell>
               {hasOnDemandColumn && (
                 <Table.Cell>
-                  {file.availability?.["on demand"] && (
+                  {isOnDemand(file) && (
                     <Popup
-                        content={file.availability.online ? "Some files of the dataset are available for immediate download" : "The files have to be requested before they are available"}
-                        trigger={<div className="ui mini message">{file.availability.online ? "Partial" : "On demand"}</div>}
-                        position="top center"
+                      content={
+                        typeof file.availability === "object" && file.availability.online
```

Comment:
```suggestion
                         file.availability?.online
```
I guess you want to know if online is a true value